File: ACM/greedy/p5585. 거스름돈.py
start_exchange_money = 500          # 거스름돈 최대치 금액 설정
money = 1000 - int(input())

is_switch = False
result = 0

while(start_exchange_money):
    exchange_cnt = money // start_exchange_money    # 잔돈 개수
    money = money - (start_exchange_money * exchange_cnt)
    result += exchange_cnt

    is_switch = not is_switch
    if (is_switch):
        start_exchange_money = start_exchange_money // 5
    else:
        start_exchange_money = start_exchange_money // 2

print(result)

File: ACM/greedy/p5585. 거스름돈.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int money = 1000 - Integer.parseInt(br.readLine());
		int startExchangeMoney = 500;
		
		boolean isSwitch = false;
		long result = 0;
		
		while (startExchangeMoney > 0) {
			int exchangeCnt = money / startExchangeMoney;
			money = money - (startExchangeMoney * exchangeCnt);
			result += exchangeCnt;
			
			isSwitch = !isSwitch;
			if (isSwitch) {
				startExchangeMoney /= 5;
			} else {
				startExchangeMoney /= 2;
			}
		}
		
		System.out.println(result);
	}
}
